src/Common/Provider/Controller.php

```diff
@@ -0,0 +1,135 @@
+<?php
+/**
+ * The base clas all Controllers should extend.
+ *
+ * @since   TBD
+ *
+ * @package TEC\Common\Provider;
+ */
+
+namespace TEC\Common\Provider;
+
+use tad_DI52_ServiceProvider as Service_Provider;
+use TEC\Common\StellarWP\ContainerContract\ContainerInterface;
+use Tribe__Log as Log;
+
+/**
+ * Class Controller.
+ *
+ * @since   TBD
+ *
+ * @package TEC\Common\Provider;
+ *
+ * @property ContainerInterface $container
+ */
+abstract class Controller extends Service_Provider {
+	/**
+	 * Registers the filters and actions hooks added by the controller if the controller has not registered yet.
+	 *
+	 * @since TBD
+	 *
+	 * @return void
+	 */
+	public function register() {
+		/*
+		 * Look up and set the value in the container request cache to allow building the same Controller
+		 * with a **different** container. (e.g. in tests).
+		 */
+		if ( $this->container->getVar( static::class . '_registered' ) ) {
+			return;
+		}
+
+		// Register the controller as a singleton in the container.
+		// @todo remove when the Container is updated to bind Providers as singletons by default.
+		$this->container->singleton( static::class, $this );
+
+		if ( ! $this->is_active() ) {
+			return;
+		}
+
+		$this->container->setVar( static::class . '_registered', true );
+
+		$this->do_register();
+	}
+
+	/**
+	 * Registers the filters and actions hooks added by the controller.
+	 *
+	 * @since TBD
+	 *
+	 * @return void
+	 */
+	abstract protected function do_register(): void;
+
+	/**
+	 * Removes the filters and actions hooks added by the controller.
+	 *
+	 * Bound implementations should not be removed in this method!
+	 *
+	 * @since TBD
+	 *
+	 * @return void Filters and actions hooks added by the controller are be removed.
+	 */
+	abstract public function unregister(): void;
+
+	/**
+	 * Whether the controller is active or not.
+	 *
+	 * Controllers will be active by default, if that is not the case, the controller should override this method.
+	 *
+	 * @since TBD
+	 *
+	 * @return bool Whether the controller is active or not.
+	 */
+	public function is_active(): bool {
+		return true;
```

Comment:
Is this a different use case than `$this->container->getVar( static::class . '_registered' )` ? Seems like this would be checking if it is registered or not...?

Reply:
This is a placeholder for children to define what constitutes "active"

Reply:
See my answer to the other questions and the difference between `is_active` and `is_registered`.